### PR TITLE
feat: UDF migrates to Function

### PR DESCRIPTION
**Pull request by jiashenC**

### CODE REVIEW

**Comment by jiashenC** [comment]
Cover migration in the code base, but also need some help to double-check migration on the documentation.

**Comment by xzdandy** [comment]
This may also breaks the link in documentations and tutorials (example: https://github.com/georgia-tech-db/evadb/blob/staging/tutorials/03-emotion-analysis.ipynb). We need to double check.

File: docs/source/reference/udfs/custom.rst [hunk 1]
```diff
@@ -1,58 +1,58 @@
 User-Defined Functions
 ======================
 
-This section provides an overview of how you can create and use a custom user-defined function (UDF) in your queries. For example, you could write an UDF that wraps around your custom PyTorch model.
+This section provides an overview of how you can create and use a custom user-defined function (Function) in your queries. For example, you could write an Function that wraps around your custom PyTorch model.
```
Comment (jarulraj) [inline docs/source/reference/udfs/custom.rst:4]: No need for `(Function)`. Please change `Function` to `function`.

File: docs/source/reference/udfs/custom.rst [hunk 2]
```diff
@@ -1,58 +1,58 @@
...
 
-Part 1: Writing a custom UDF
+Part 1: Writing a custom Function
 ------------------------------
 
-During each step, use `this UDF implementation <https://github.com/georgia-tech-db/evadb/blob/master/evadb/udfs/yolo_object_detector.py>`_  as a reference.
+During each step, use `this Function implementation <https://github.com/georgia-tech-db/evadb/blob/master/evadb/functions/yolo_object_detector.py>`_  as a reference.
```
Comment (jarulraj) [inline docs/source/reference/udfs/custom.rst:9]: see `this function's implementation

File: docs/source/reference/udfs/custom.rst [hunk 3]
```diff
@@ -1,58 +1,58 @@
...
 Setup
 -----
 
-An abstract method that must be implemented in your child class. The setup function can be used to initialize the parameters for executing the UDF. The parameters that need to be set are 
+An abstract method that must be implemented in your child class. The setup function can be used to initialize the parameters for executing the Function. The parameters that need to be set are 
 
 - cacheable: bool
  
-  - True: Cache should be enabled. Cache will be automatically invalidated when the UDF changes.
+  - True: Cache should be enabled. Cache will be automatically invalidated when the Function changes.
```
Comment (jarulraj) [inline docs/source/reference/udfs/custom.rst:30]: function's implementation code changes

File: docs/source/reference/udfs/custom.rst [hunk 4]
```diff
@@ -111,39 +111,39 @@ A sample forward function is given below
...
+        * <path_to_implementation> - specifies the path to the implementation class for the Function
 
-  Here, is an example query that registers a UDF that wraps around the 'YoloObjectDetection' model that performs Object Detection.
+  Here, is an example query that registers a Function that wraps around the 'YoloObjectDetection' model that performs Object Detection.
 
   .. code-block:: sql
 
-    CREATE UDF YoloDecorators
-    IMPL  'evadb/udfs/decorators/yolo_object_detection_decorators.py';
+    CREATE FUNCTION YoloDecorators
```
Comment (jarulraj) [inline docs/source/reference/udfs/custom.rst:133]: Do we need `Decorators` suffix still?

File: docs/source/reference/udfs/custom.rst [hunk 5]
```diff
@@ -111,39 +111,39 @@ A sample forward function is given below
...
+    IMPL  'evadb/functions/decorators/yolo_object_detection_decorators.py';
     
 
-  A status of 0 in the response denotes the successful registration of this UDF.
+  A status of 0 in the response denotes the successful registration of this Function.
 
-2. Now you can execute your UDF on any video:
+2. Now you can execute your Function on any video:
```
Comment (jarulraj) [inline docs/source/reference/udfs/custom.rst:139]: function

File: docs/source/reference/udfs/custom.rst [hunk 6]
```diff
@@ -111,39 +111,39 @@ A sample forward function is given below
...
 
   .. code-block:: sql
 
       SELECT YoloDecorators(data) FROM MyVideo WHERE id < 5;
 
-3. You can drop the UDF when you no longer need it.
+3. You can drop the Function when you no longer need it.
```
Comment (jarulraj) [inline docs/source/reference/udfs/custom.rst:145]: function

File: docs/source/reference/udfs/custom.rst [hunk 7]
```diff
@@ -111,39 +111,39 @@ A sample forward function is given below
...
   .. code-block:: sql
 
-    CREATE UDF YoloDecorators
-    IMPL  'evadb/udfs/decorators/yolo_object_detection_decorators.py';
+    CREATE FUNCTION YoloDecorators
+    IMPL  'evadb/functions/decorators/yolo_object_detection_decorators.py';
     
 
-  A status of 0 in the response denotes the successful registration of this UDF.
+  A status of 0 in the response denotes the successful registration of this Function.
```
Comment (jarulraj) [inline docs/source/reference/udfs/custom.rst:137]: function

File: evadb/README.md [hunk 8]
```diff
@@ -10,6 +10,6 @@
 * SELECT statements are expanded to different operators PROJECT and FILTER, etc.
 * `optimizer / plan_generator.py` - Generation of query plan.
 * All operators in the tree are converted to group expression for optimization(`optimizer / group_expression.py`).
-* The optimizer rewrites the query based on a collection of UDF-centric rules(`optimizer / rules`).
+* The optimizer rewrites the query based on a collection of Function-centric rules(`optimizer / rules`).
```
Comment (jarulraj) [inline evadb/README.md:13]: rules for optimizing AI functions

**Review by jarulraj — APPROVED** [review]

**Review by xzdandy — APPROVED** [review]

**Pull request merged**